File: src/main/java/no/nav/veilarboppfolging/controller/request/VeilederTilordning.java
package no.nav.veilarboppfolging.controller.request;

import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
public class VeilederTilordning {
    String brukerFnr;
    String aktoerId;
    String innloggetVeilederId;
    String fraVeilederId;
    String tilVeilederId;
}

File: src/main/kotlin/no/nav/veilarboppfolging/controller/request/VeilederTilordning.kt
package no.nav.veilarboppfolging.controller.request

import no.nav.veilarboppfolging.controller.response.Veileder

data class VeilederTilordning(
    val brukerFnr: String,
    val aktoerId: String?,
    val innloggetVeilederId: String?,
    val fraVeilederId: String?,
    val tilVeilederId: String
) {
    // Hjelpefunksjoner for å lette migrering vekk fra Lombok
    fun oppdaterMedInnloggetVeilederId(veilederId: String) = this.copy(innloggetVeilederId = veilederId)
    fun oppdaterMedAktorId(nyAktorId: String) = this.copy(aktoerId = nyAktorId)
}
